### ISSUE TITLE
GWAS-by-subtraction: extraordinary high GWAS p-value from the outputGWAS()

### ISSUE THREAD

**Issue by WeiCSong** [GenomicSEM/GenomicSEM#38]
Hi, 
    I ran a GWAS-by-subtraction and the result from LDSC and usermodel() seems reasonable (see below). But the userGWAS returned very high p-value: the lowest p-value for >5m SNPs was 0.05, which definitely violates a uniform distribution:

> head(d[order(d[,15]),])
                 SNP CHR        BP       MAF A1 A2 lhs op rhs free label
57837338    rs836952   9  25904637 0.0685885  C  T ncm  ~ SNP    5      
57709235  rs79300279   9   6133663 0.0407555  C  A ncm  ~ SNP    5      
57837336   rs1219935   9  25902763 0.0755467  C  T ncm  ~ SNP    5      
5889791   rs11198504  10 120368534 0.0536779  T  G ncm  ~ SNP    5      
58897836  rs11198501  10 120367850 0.0536779  G  C ncm  ~ SNP    5      
512627839  rs3865368  18  11974749 0.0695825  T  G ncm  ~ SNP    5      
                 est         SE Z_Estimate Pval_Estimate        chisq
57837338  -0.1268818 0.06668243  -1.902776    0.05706972 3.432989e-17
57709235  -0.1427829 0.07660701  -1.863835    0.06234482 1.949406e-16
57837336  -0.1174886 0.06424197  -1.828846    0.06742272 1.225387e-15
5889791   -0.1270428 0.07218130  -1.760052    0.07839900 9.414978e-18
58897836  -0.1269866 0.07218207  -1.759254    0.07853435 2.210379e-16
512627839 -0.1137829 0.06561039  -1.734221    0.08287881 2.639530e-16
          chisq_df chisq_pval AIC error warning
57837338         0          0  12     0       0
57709235         0          0  12     0       0
57837336         0          0  12     0       0
5889791          0          0  12     0       0
58897836         0          0  12     0       0
512627839        0          0  12     0       0

It is acceptable that the latent factor ncm does not have a significant heritability, but i suppose such high p-value cannot be explained by random effect. Could you provide some suggestion on the issue? Thank you very much for your help.

My usermodel() model and outcome:
![image](https://user-images.githubusercontent.com/41194740/120874856-28ec0c00-c5db-11eb-9ef9-4b277487149c.png)

> output
$modelfit
   chisq df p_chisq AIC CFI SRMR
df    NA  0      NA  NA  NA   NA
$results
  lhs op rhs Unstand_Est          Unstand_SE STD_Genotype    STD_Genotype_SE
5  cm =~ MDD   0.1089702 0.00816797489531124    0.6697152 0.0501992078896497
4  cm =~  CM   0.2379085  0.0101221997048112    1.0000000 0.0425466059304789
9 ncm =~ MDD   0.1208323 0.00812108043689905    0.7426180 0.0499110012486965
8 ncm ~~ ncm   1.0000000                        1.0000000                   
1  cm ~~  cm   1.0000000                        1.0000000                   
3  cm ~~ ncm   0.0000000                        0.0000000                   
6 MDD ~~  CM   0.0000000                        0.0000000                   
2  CM ~~  CM   0.0000000                        0.0000000                   
7 MDD ~~ MDD   0.0000000                        0.0000000                   
    STD_All       p_value
5 0.6697152  1.333828e-40
4 1.0000000 3.743973e-122
9 0.7426180  4.522471e-50
8 1.0000000            NA
1 1.0000000            NA
3 0.0000000            NA
6 0.0000000            NA
2 0.0000000            NA
7 0.0000000            NA

my userGWAS model: 
![image](https://user-images.githubusercontent.com/41194740/120875076-7ae16180-c5dc-11eb-94a0-7cf0024f3d9b.png)

my LDSC output: 

> Estimating heritability [1/3] for: CM.sumstats.gz
Heritability Results for trait: CM.sumstats.gz
Mean Chi^2 across remaining SNPs: 1.1537
Lambda GC: 1.1375
Intercept: 1.0115 (0.0075)
Ratio: 0.0746 (0.0489)
Total Observed Scale h2: 0.0566 (0.0048)
h2 Z: 11.751823     
Calculating genetic covariance [2/3] for traits: CM.sumstats.gz and MDD.sumstats.gz
1058952 SNPs remain after merging CM.sumstats.gz and MDD.sumstats.gz summary statistics
Results for genetic covariance between: CM.sumstats.gz and MDD.sumstats.gz
Mean Z*Z: 0.1481
Cross trait Intercept: 0.0158 (0.0062)
Total Observed Scale Genetic Covariance (g_cov): 0.0259 (0.0021)
g_cov Z: 12.553063
g_cov P-value: 3.8242e-36          
Estimating heritability [3/3] for: MDD.sumstats.gz
Heritability Results for trait: MDD.sumstats.gz
Mean Chi^2 across remaining SNPs: 1.2522
Lambda GC: 1.2242
Intercept: 0.9935 (0.0081)
Ratio: -0.0258 (0.0322)
Total Observed Scale h2: 0.0265 (0.0015)
h2 Z: 17.105943

**Comment by MichelNivard** [comment]
Hi I am slightly confused, you say ncm has no heritability, but it does in your model right?

ncm =~ MDD 0.1208323 0.00812108043689905 0.7426180 0.0499110012486965

**Comment by MichelNivard** [comment]
did you get any per SNP errors? maybe the starting values are too far off?

**Comment by WeiCSong** [comment]
Thanks for your help! The R output seems to contain no error, i'll try some
different priors to see if the output would be as expected.

MichelNivard ***@***.***> 于2021年6月11日周五 下午8:58写道：

**Comment by MichelNivard** [comment]
Did you solve this? If soe how? so we can help others that run into the issue.

**Comment by WeiCSong** [comment]
HI, I repeated the analysis with several updates and the results have appeared right. My guess is that the essential update is on the sumstats() function: I didnt't find necessary information from the original GWAS paper so I tried several combinations of se.logit, OLS and lin.prob, and find a combination that returned a reasonable output p-value. Other updates like different priors and sample sizes seem to have little impact.
